### PR TITLE
Feat(Fake Ads):Ads Management

### PR DESCRIPTION
**Pull request by TuyishimireEric**
#### What does this PR do?

Fake Ads management

#### Description of Task to be completed?

Add a docker service in our docker network to host a node.js middleware . this will host a serveless framework that runs a AWS lamda function / functions to perform ads management

#### How should this be manually tested?

- download and install `Docker Desktop`
- run `docker-compose build`
- run `docker-compose up -d`
- check the [swagger documentation ](http://localhost:3000/swagger/)

#### Any background context you want to provide?

-dummy data are provided before having the real data

#### What are the relevant pivotal tracker/Trello stories?

- id of the project #187535721

### CODE REVIEW

**Review by fab-ryan — CHANGES_REQUESTED** [review]
Please first rebase

File: Docker-compose.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: '3.8'
 
 services:
   db:
-    image: postgres
+    image: tuyishimireeric/postgres
```
Comment (fab-ryan) [inline Docker-compose.yml]: why this image name please provide apropriate name

File: src/config/config.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const dotenv = require('dotenv');
 dotenv.config();
 
 const getPrefix = () => {
-  const env = process.env.NODE_ENV || 'production';
+  const env = process.env.NODE_ENV || 'development';
   const envPrefixMap = {
```
Comment (fab-ryan) [inline src/config/config.js]: suddenly changing this line

File: src/controllers/productController.ts [hunk 3]
```diff
@@ -220,12 +220,7 @@ export const getAProduct = async (req: Request, res: Response) => {
       return sendResponse<null>(res, 404, null, 'Product not found!');
     }
 
-    return sendResponse<ProductAttributes>(
-      res,
-      200,
-      product,
-      'Product found successfully!'
-    );
+    return sendResponse(res, 200, product, 'Product found successfully!');
```
Comment (fab-ryan) [inline src/controllers/productController.ts]: why EsLint not applied here

File: src/routers/productRoute.ts [hunk 4]
```diff
@@ -48,4 +49,8 @@ router.patch(
 
 router.delete('/products/:id', isAuthenticated, isSeller, deleteAProduct);
 
+router.get('/ads', getAds);
```
Comment (fab-ryan) [inline src/routers/productRoute.ts]: please add validation for queries using the same middleware

File: startup.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+export PGADMIN_SERVERS='{"Servers": [{"Name": "My Database", "Group": "Servers", "Host": "'$DATABASE_HOST'", "Port": "'$DATABASE_PORT'", "MaintenanceDB": "'$DATABASE_NAME'", "Username": "'$DATABASE_USER'", "Password": "'$DATABASE_PASSWORD'", "SSLMode": "prefer"}]}'
+
+echo "[general]" > /etc/pgadmin/pgadmin4.conf
+echo "data_dir = /var/lib/pgadmin" >> /etc/pgadmin/pgadmin4.conf
+echo "logs_dir = /var/log/pgadmin" >> /etc/pgadmin/pgadmin4.conf
+
+echo "$PGADMIN_SERVERS" > /etc/pgadmin/pgadmin4.conf
+
+exec /usr/bin/pgadmin4
```
Comment (fab-ryan) [inline startup.sh]: what is the difference between with this file and current one???
if it's PG admin setup/config you can add theme in that file

**Review by fab-ryan — CHANGES_REQUESTED** [review]
Work on those changes

File: src/config/config.js (hunk 2 above)
Comment (TuyishimireEric) [reply src/config/config.js]: Do I have to keep the production?

Comment (fab-ryan) [reply src/config/config.js]: yes

File: src/controllers/productController.ts (hunk 3 above)
Comment (TuyishimireEric) [reply src/controllers/productController.ts]: Solved

File: Docker-compose.yml (hunk 1 above)
Comment (TuyishimireEric) [reply Docker-compose.yml]: Resolved,

**Review by fab-ryan — APPROVED** [review]

**Review by Mukunzijames — APPROVED** [review]

**Review by RWEMAREMY — APPROVED** [review]
Keep it up @TuyishimireEric 💯

**Review by Sophie20202 — APPROVED** [review]

**Pull request merged**